### PR TITLE
[addondb/addonrepos] pure cleanup and vector capacity reservations

### PR DESCRIPTION
**Pull request by howie-f**
## Description
as the title says.. all commits buildable. so this should be a fun review 

## How has this been tested?
running on live system for quite some time

## Types of change

- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document

### CODE REVIEW

File: xbmc/addons/AddonDatabase.cpp [hunk 1]
```diff
@@ -634,14 +583,14 @@ bool CAddonDatabase::GetAddon(const std::string& addonID, const AddonVersion& ve
     if (!m_pDS)
       return false;
 
-  std::string sql = PrepareSQL(
-      "SELECT addons.id FROM addons "
-      "JOIN addonlinkrepo ON addonlinkrepo.idAddon=addons.id "
-      "JOIN repo ON repo.id=addonlinkrepo.idRepo "
-      "WHERE addons.addonID='%s' AND addons.version='%s' AND repo.addonID='%s'",
-      addonID.c_str(), version.asString().c_str(), repoId.c_str());
+    std::string sql =
```
Comment (AlwinEsch) [inline xbmc/addons/AddonDatabase.cpp:586]: Maybe change direct to `const std::string` as them no more becomes changed and more clear for eyes and maybe compiler.

**Review by AlwinEsch — APPROVED** [review]
Except the note about `const` looks it good to me.

File: xbmc/addons/AddonDatabase.cpp (hunk 1 above)
Comment (howie-f) [reply xbmc/addons/AddonDatabase.cpp]: sure, done :+1:

**Comment by AlwinEsch** [comment]
Thanks for the change.

Sometimes strange what `clang` in Jenkins want and makes no sense :thinking:.

**Comment by howie-f** [comment]
seems that clang-format also formats context of the diffs. and the two changes were so close together. squashed them and all good

**Pull request merged**